packages/components/src/ContextualHelp/ContextualHelp.tsx

```diff
@@ -0,0 +1,155 @@
+import { ReactNode, forwardRef } from 'react';
+import { Dialog, DialogTrigger, Popover } from 'react-aria-components';
+import { useClassNames } from '@marigold/system';
+import { Button } from '../Button';
+
+// Props
+// ---------------
+/**
+ * Variant of the contextual help icon.
+ * @default help
+ */
+type Variant = 'help' | 'info';
+
+/**
+ * Size of the trigger button.
+ * @default XS
+ */
+type Size = 'XS' | 'S';
+
+/**
+ * Placement of the popover.
+ * @default bottom start
+ */
+type Placement =
+  | 'top'
+  | 'bottom'
+  | 'left'
+  | 'right'
+  | 'top start'
+  | 'bottom start';
+
+/**
+ * Props for the ContextualHelp component.
+ */
+export interface ContextualHelpProps {
+  /** Content rendered inside the popover. */
+  children: ReactNode;
+
+  /** Visual variant of the icon (e.g. info or help). */
+  variant?: Variant;
+
+  /** Size of the trigger button. */
+  size?: Size;
+
+  /** Placement of the popover relative to the button. */
+  placement?: Placement;
+
+  /** Offset (in px) between button and popover. */
+  offset?: number;
+
+  /** Whether the popover is open by default (uncontrolled). */
+  defaultOpen?: boolean;
+
+  /** Controls the open state of the popover (controlled). */
+  isOpen?: boolean;
```

Comment:
open instead of `isOpen`, see https://www.marigold-ui.io/__internal__/component-guidelines?theme=core#props-naming-convention :)